kiva/tests/test_graphics_context.py

```diff
@@ -4,8 +4,18 @@
     alltrue, array, concatenate, dtype,
     frombuffer, newaxis, ravel, ones, zeros)
 from PIL import Image as PILImage
-from hypothesis import given
-from hypothesis.strategies import sampled_from
+try:
+    from hypothesis import given
+    from hypothesis.strategies import sampled_from
+    hypothesis_available = True
+except ModuleNotFoundError:
+    def given(_):
+        def func(_):
+            pass
+        return func
+    def sampled_from(_):
+        pass
```

Comment:
this was a bit awkward, but because `given` and `sampled_from` are used in decorators, so even if I try to skip those tests when `hypothesis_available == False` the tests still fail when trying to load them with something like:
```
======================================================================
ERROR: kiva.tests.test_graphics_context (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: kiva.tests.test_graphics_context
Traceback (most recent call last):
  File "/Users/aayres/.edm/envs/enable-test-3.6-null-pillow/lib/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/Users/aayres/.edm/envs/enable-test-3.6-null-pillow/lib/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/Users/aayres/Desktop/enable/kiva/tests/test_graphics_context.py", line 27, in <module>
    class TestAlphaBlackImage(unittest.TestCase):
  File "/Users/aayres/Desktop/enable/kiva/tests/test_graphics_context.py", line 33, in TestAlphaBlackImage
    @given(sampled_from([1.0, 0.0, 0.5]))
NameError: name 'given' is not defined
```

There is probably a cleaner more standard way to do this sort of thing?  I am not sure

Reply:
`hypothesis` is a test dependency. I am not sure we need to skip these. We should make sure the test dependency is included in the `setup.py` though. CI does have `hypothesis` installed and these tests are run there.

Reply:
To elaborate on why I am not sure we need to skip tests based on missing test dependencies: TraitsUI has `packaging` as its test dependency, and it was specified in `extras_require`. Whenever I do `pip install traitsui` and try to run the test suite, I am greeted with an import error due to missing `packaging`: The tests requiring `packaging` are not skipped because `packaging` is not available. That import error reminds me to reinstall with `pip install traitsui[test]`.  If the test suite was permissive about missing test dependencies, then I risk forgetting to install those dependencies and end up running much fewer tests.

Test dependencies are still separated from optional dependencies due to optional features of the tested system (e.g. ArrayEditor is an optional feature of TraitsUI that depends on NumPy). When a package has optional features with optional dependencies, it is expected to be able to run the tests in the absence of those dependencies.